website/frontend/src/components/pages/SendMessage/SendMessage.tsx

```diff
@@ -1,46 +1,54 @@
-import React, { useCallback, useEffect, useState, VFC } from 'react';
-import { useDispatch } from 'react-redux';
+import React, { useEffect, useState, VFC } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
 import { useNavigate, useParams } from 'react-router-dom';
 import { Metadata, createPayloadTypeStructure, decodeHexTypes } from '@gear-js/api';
 import { MetaParam } from 'utils/meta-parser';
-import { RPC_METHODS } from 'consts';
-import ServerRPCRequestService, { RPCResponseError } from 'services/ServerRPCRequestService';
-import { GetMetaResponse } from 'api/responses';
+import { RPCResponseError } from 'services/ServerRPCRequestService';
 import { EventTypes } from 'types/alerts';
-import { AddAlert } from 'store/actions/actions';
+import { programService } from 'services/ProgramsRequestService';
+import { RootState } from 'store/reducers';
+import { AddAlert, getMessageAction } from 'store/actions/actions';
 import { isDevChain, getLocalProgramMeta, fileNameHandler } from 'helpers';
 import { MessageForm } from './children/MessageForm/MessageForm';
 import ArrowBack from 'assets/images/arrow_back.svg';
 import ProgramIllustration from 'assets/images/program_icon.svg';
 import { Spinner } from 'components/blocks/Spinner/Spinner';
 import './SendMessage.scss';
 
+type Params = {
+  programId: string;
+  messageId: string;
+};
+
 export const SendMessage: VFC = () => {
   const dispatch = useDispatch();
   const navigate = useNavigate();
-  const routeParams = useParams();
-  const programId = routeParams.id as string;
+  const { programId = '', messageId = '' } = useParams<Params>();
```

Comment:
```suggestion
  const { programId = '', messageId = '' } = useParams();
```
no need for explicit type